src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1356,7 +1356,11 @@ static void copy(MemorySegment srcSegment, ValueLayout srcElementLayout, long sr
         }
         long size = elementCount * srcElementLayout.byteSize();
         srcImpl.checkAccess(srcOffset, size, true);
-        dstImpl.checkAccess(dstOffset, size, false);
+        if (dstImpl instanceof NativeMemorySegmentImpl && dstImpl.byteSize() == 0) {
```

Comment:
As Jorn said, this change is not acceptable, as it allows bulk copy disregarding the segment real size. In such cases, the issue is always a missing unsafe resize somewhere in the linker code.

Reply:
Removed this workaround. I'm glad to get rid of it.